pkg/util/kubernetes/apiserver/apiserver.go

```diff
@@ -443,3 +398,26 @@ func (c *APIClient) GetRESTObject(path string, output runtime.Object) error {
 
 	return result.Into(output)
 }
+
+// StartMetadataController runs the metadata controller to collect cluster metadata. This is
+// only called once, when we have confirmed we could correctly connect to the API server.
+func StartMetadataController(clientBuilder *ClientBuilder, stopCh chan struct{}) error {
```

Comment:
This is purposely not a member function of `APIClient` so that it's clear that it uses a different underlying `rest.Client`.

The one awkward thing that remains is we are using a different `rest.Client` instance for the `connect()` logic. If the different `rest.Client` we have diverge in a meaningful way this could cause bugs that will be hard to find. Right now the only difference is the `rest.Config.Timeout`.